### PR TITLE
Fish light attraction

### PR DESCRIPTION
**Pull request by stellarxo**
It works!!
Right now, checks if player IsVisible() by checking if the light toggle is switched on.

Maybe PlayerLightToggle should be switched into LightToggle so all characters can use it?

Also, changed LightSource FixedUpdate() to just Update() because it was being really laggy.

### CODE REVIEW

File: Assets/Scripts/Player/PlayerLightToggle.cs [hunk 1]
```diff
@@ -1,7 +1,7 @@
 using UnityEngine;
 using System.Collections;
 
-public class PlayerLightToggle
+public class PlayerLightToggle : MonoBehaviour
```
Comment (alisterdev) [inline Assets/Scripts/Player/PlayerLightToggle.cs]: Why extend Mono? It was meant to be a pure C# class

File: Assets/Scripts/Player/PlayerLightToggle.cs [hunk 2]
```diff
@@ -49,4 +48,19 @@ public void LostOfLight(float lostOfLightTimeInterval, float energyCostLightTogg
...
+    public bool IsVisible()
+    {
+        if (this.lightsEnabled)
+        {
+            return true;
+        }
+        else
+        {
+            return false;
+        }
```
Comment (alisterdev) [inline Assets/Scripts/Player/PlayerLightToggle.cs]: What's the point of this? You can juste create a getter for lightsEnabled

File: Assets/Scripts/Player/Player.cs [hunk 3]
```diff
@@ -137,5 +131,12 @@ public void OnLightDepleted()
...
     }
-
-}
\ No newline at end of file
+    
+    /// <summary>
+    /// If player lights are on, player is visible
+    /// </summary>
+    public bool IsVisible()
+    {
```
Comment (alisterdev) [inline Assets/Scripts/Player/Player.cs]: Player class has an instance of lightToggle in it. you can use it where needed, and get lightToggle status through a getter

File: Assets/Scripts/Player/Player.cs [hunk 4]
```diff
@@ -41,7 +41,7 @@ public class Player : LightSource
 
     /** Caches the player's components */
     private PlayerMovement movement;
-    private PlayerLightToggle lightToggle;
+    public PlayerLightToggle lightToggle;
```
Comment (alisterdev) [inline Assets/Scripts/Player/Player.cs]: no-no

File: Assets/Scripts/Light/LightSource.cs [hunk 5]
```diff
@@ -31,7 +31,7 @@ public virtual void Awake()
         lightEnergy = new LightEnergy(defaultEnergy, gameObject, debugInfiniteLight);
     }
 
-    public virtual void FixedUpdate()
+    public virtual void Update()
```
Comment (alisterdev) [inline Assets/Scripts/Light/LightSource.cs]: @johnny-la critics/compliments of that is on you

**Comment by alisterdev** [comment]
@gogo76 wonder if that notifies you on slack

File: Assets/Scripts/Light/LightSource.cs (hunk 5 above)
Comment (johnny-la) [inline Assets/Scripts/Light/LightSource.cs]: As long as you multiply framerate-dependent operations by `Time.deltaTime` in `Update()`, it should be fine. The point of `FixedUpdate()` is to allow operations to run at the same framerate as physics. Whether or not you'd want that is dependent on the specific operation you're performing and its use-case.

File: Assets/Scripts/Player/Player.cs (hunk 4 above)
Comment (stellarxo) [inline Assets/Scripts/Player/Player.cs]: oops sorry did this for debugging

File: Assets/Scripts/Player/PlayerLightToggle.cs (hunk 2 above)
Comment (stellarxo) [inline Assets/Scripts/Player/PlayerLightToggle.cs]: that's what this was supposed to be

File: Assets/Scripts/Player/PlayerLightToggle.cs (hunk 1 above)
Comment (stellarxo) [inline Assets/Scripts/Player/PlayerLightToggle.cs]: was getting an error about it not extending monobehaviour or being an interface

**Pull request closed without merging**